REGRESSION(268320@main): WebDriver fullscreen/input type=file tests failing

gsnedders opened the pull request:
#### 6444dd094f2320ce8f12512bea72a07f942387fc
<pre>
REGRESSION(268320@main): WebDriver fullscreen/input type=file tests failing
<a href="https://bugs.webkit.org/show_bug.cgi?id=263887">https://bugs.webkit.org/show_bug.cgi?id=263887</a>
<a href="https://rdar.apple.com/117683701">rdar://117683701</a>

Reviewed by Tim Nguyen.

Partially revert 268320@main (maintaining the exported WPT test changes,
but reverting all actual source changes) to return to the prior
behaviour of treating all automation protocol scripts as invoked via a
user gesture, fixing the regression but re-introducing the WebDriver
spec violation.

* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
(WebKit::WebAutomationSessionProxy::evaluateJavaScriptFunction):

Canonical link: <a href="https://commits.webkit.org/270201@main">https://commits.webkit.org/270201@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a2f1edf85a84358cc638192344c74da595da1baf

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/24815 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/3360 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/26069 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/26934 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/22781 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/5039 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/797 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/23109 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/25060 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/2403 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/21425 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/27517 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/2110 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/28510 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/22633 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/22702 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/26328 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/2048 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/351 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/3337 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5948 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/2498 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/2399 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas reviewed (commented):
Summarizing a conversation from elsewhere --- it's unfortunate that we have to revert to our previous behavior of treating all Automation Protocol script as backed by a user gesture, but that's because said behavior was papering over the long standing bug that we don't treat (some of) our automation script as though they were invoked by a user gesture.

I am doing some investigation into how small this subset of API that should be blessed with user activation is, as well as how much plumbing is required. If the scope is too large for our timeline, I am onboard with the partial revert done by this PR.

aprotyas reviewed (commented):
Some additional context -- I'm trying the approach of forcing user gesture for certain script execution in https://github.com/WebKit/WebKit/pull/19830. If there's too much scope creep there or we can't merge it within our timeline constraints, we should merge this PR instead.

aprotyas commented on Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
```diff
@@ -421,7 +421,10 @@ void WebAutomationSessionProxy::evaluateJavaScriptFunction(WebCore::PageIdentifi
 
-    callPropertyFunction(context, scriptObject, "evaluateJavaScriptFunction"_s, std::size(functionArguments), functionArguments, &exception);
+    {
+        WebCore::UserGestureIndicator gestureIndicator(WebCore::ProcessingUserGesture, frame->coreLocalFrame()->document());
```
```suggestion
        WebCore::UserGestureIndicator gestureIndicator(WebCore::IsProcessingUserGesture::Yes, frame->coreLocalFrame()->document());
```

gsnedders replied:
Sorry, I forgot to commit and push that change!

gsnedders committed:
REGRESSION(268320@main): WebDriver fullscreen/input type=file tests failing
https://bugs.webkit.org/show_bug.cgi?id=263887
rdar://117683701

Reviewed by Tim Nguyen.

Partially revert 268320@main (maintaining the exported WPT test changes,
but reverting all actual source changes) to return to the prior
behaviour of treating all automation protocol scripts as invoked via a
user gesture, fixing the regression but re-introducing the WebDriver
spec violation.

* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
(WebKit::WebAutomationSessionProxy::evaluateJavaScriptFunction):

Canonical link: https://commits.webkit.org/270201@main
